src/main/java/eu/dissco/exportjob/repository/ElasticSearchRepository.java

```diff
@@ -23,23 +23,62 @@
 @RequiredArgsConstructor
 public class ElasticSearchRepository {
 
+  private static final String SORT_BY = "dcterms:identifier.keyword";
   private final ElasticsearchClient client;
   private final ElasticSearchProperties properties;
-  private static final String SORT_BY = "dcterms:identifier.keyword";
+
+  private static List<Query> generateQuery(List<SearchParam> searchParams) {
+    var qList = new ArrayList<Query>();
+    for (var searchParam : searchParams) {
+      var key = searchParam.inputField()
+          .replace("'", "")
+          .replace("[*]", "")
+          .replace("$", "")
+          .replace("[", "")
+          .replace("]", ".")
+          .replace("\"", "")
+          + "keyword";
+      var val = searchParam.inputValue();
+      if (val == null || val.isEmpty()) {
+        qList.add(
+            new Query.Builder().bool(b -> b.mustNot(q -> q.exists(e -> e.field(key)))).build());
+      } else {
+        qList.add(
+            new Query.Builder().term(t -> t.field(key).value(val).caseInsensitive(true)).build());
+      }
+    }
+    return qList;
+  }
 
   public List<JsonNode> getTargetObjects(List<SearchParam> searchParams, TargetType targetType,
       String lastId, List<String> targetFields)
       throws IOException {
     var query = generateQuery(searchParams);
     var index = getIndex(targetType);
 
+    return retrieveObjects(lastId, properties.getPageSize(), targetFields, index, query);
+  }
+
+  public List<JsonNode> getTargetForMediaList(List<String> mediaList) throws IOException {
```

Comment:
I would call this "getMediaByID". or "getMediaTargets". current name sounds like it's getting the targets of media objects

Reply:
:+1: